test/units/config/test_manager.py

```diff
@@ -23,88 +27,90 @@
                 'config_entry_multi_deprecated_source': Setting(name='config_entry_multi_deprecated_source', value=u'morefromini',
                                                                 origin=cfg_file, type='string')}
 
-
-class TestConfigData(unittest.TestCase):
-
-    def setUp(self):
-        self.manager = ConfigManager(cfg_file, os.path.join(curdir, 'test.yml'))
-
-    def tearDown(self):
-        self.manager = None
+ensure_test_data = [
+    ('a,b', 'list', list),
+    (['a', 'b'], 'list', list),
+    ('y', 'bool', bool),
+    ('yes', 'bool', bool),
+    ('on', 'bool', bool),
+    ('1', 'bool', bool),
+    ('true', 'bool', bool),
+    ('t', 'bool', bool),
+    (1, 'bool', bool),
+    (1.0, 'bool', bool),
+    (True, 'bool', bool),
+    ('n', 'bool', bool),
+    ('no', 'bool', bool),
+    ('off', 'bool', bool),
+    ('0', 'bool', bool),
+    ('false', 'bool', bool),
+    ('f', 'bool', bool),
+    (0, 'bool', bool),
+    (0.0, 'bool', bool),
+    (False, 'bool', bool),
+    ('10', 'int', int),
+    (20, 'int', int),
+    ('0.10', 'float', float),
+    (0.2, 'float', float),
+    ('/tmp/test.yml', 'pathspec', list),
+    ('/tmp/test.yml,/home/test2.yml', 'pathlist', list),
+    ('a', 'str', str),
+    ('a', 'string', str),
+    ('None', 'none', type(None))
+]
```

Comment:
wont int/float/str have issues across python versions?

Reply:
Good catch, I will check and modify this code.